api/turing/imagebuilder/imagebuilder.go

```diff
@@ -486,3 +560,50 @@ func (ib *imageBuilder) DeleteImageBuildingJob(
 	err = ib.clusterController.DeleteJob(context.Background(), ib.imageBuildingConfig.BuildNamespace, job.Name)
 	return err
 }
+
+func (ib *imageBuilder) GetEnsemblerImage(
+	project *mlp.Project,
+	ensembler *models.PyFuncEnsembler,
+) (EnsemblerImage, error) {
+	imageName := ib.nameGenerator.generateDockerImageName(project.Name, ensembler.Name)
+	imageExists, err := ib.checkIfImageExists(imageName, ensembler.RunID)
+	if err != nil {
+		return EnsemblerImage{}, err
+	}
+
+	imageRef := fmt.Sprintf("%s:%s", imageName, ensembler.RunID)
+
+	image := EnsemblerImage{
+		ProjectID:           models.ID(project.ID),
+		EnsemblerID:         models.ID(ensembler.GetID()),
+		EnsemblerRunnerType: ib.runnerType,
```

Comment:
I can't tell whether it's cleaner to reuse the existing `ib.nameGenerator` field to perform the check `_, isJobType := ib.nameGenerator.(*ensemblerJobNameGenerator)` to find out the type (service/job) of the image builder **_or_**, to introduce a new field `runnerType` to the `imageBuilder` struct and then potentially remove all the `nameGenerator`-derived structs (ensemblerServiceNameGenerator/ensemblerJobNameGenerator) because they can use the new `runnerType` field to determine which naming conventions to use. 

Still sitting on the fence on this for now so what do you all think?

Reply:
I agree that there should be a better design pattern for this to simplify the nameGenerator and runnerType while at the same time make it consistent. However, I think we should revisit this in separate PR. I'm resolving this comment for now 🙏🏻

Reply:
Sounds good! Thanks!